frontend/src/types/index.ts

```diff
@@ -103,3 +98,24 @@ export interface DirectoryData {
     tooling: Tools;
   };
 }
+
+export enum DatabaseType {
+  Dynamo,
+  Disk,
+}
+
+export interface Database {
+  type: DatabaseType;
+  init: () => Promise<void>;
+  storeProject: (projectInfo: Result) => Promise<string>;
+  updateProject: (projectId: string, boardLink: string) => Promise<string>;
+  getProject: (projectId: string) => Promise<DatabaseModel>;
```

Comment:
Similarly, you can name the methods generically, `storeProject -> create`, `updateProject -> update`, `getProject -> find` are common names for these operations